test/integ/pandas/test_arrow_pandas.py

```diff
@@ -567,6 +569,118 @@ def test_timestampltz(conn_cnx, scale, timezone):
     finish(conn_cnx, table)
 
 
+def do_validate_pandas(
+    cnx_table,
+    sql,
+    cases,
+    col_count,
+    method="one",
+    data_type="float",
+    epsilon=None,
+    scale=0,
+    timezone=None,
+    telemetry_test=None,
+    telemetry_to_check=None,
+):
+    row_count = len(cases)
+    assert col_count != 0, "# of columns should be larger than 0"
+
+    cursor_table = cnx_table.cursor()
+    cursor_table.execute(SQL_ENABLE_ARROW)
+    cursor_table.execute(sql)
+
+    # build dataframe
+    total_rows, total_batches = 0, 0
+    start_time = time.time()
+
+    if method == "one":
+        df_new = cursor_table.fetch_pandas_all()
+        total_rows = df_new.shape[0]
+    else:
+        for df_new in cursor_table.fetch_pandas_batches():
+            total_rows += df_new.shape[0]
+            total_batches += 1
+
+    if telemetry_to_check is not None and telemetry_test is not None:
+        assert any(
+            t.message["type"] == telemetry_to_check for t in telemetry_test.records
+        )
+    end_time = time.time()
```

Comment:
This used to be right after the fetch calls. Should we put this before looking through the telemetry records?

Reply:
end_time? Agree.